File: server/src/main/java/com/mbpolan/retrorealms/beans/info/ServerInfo.java
package com.mbpolan.retrorealms.beans.info;

/**
 * Bean that contains information about assets and other game data about this server.
 *
 * @author mbpolan
 */
public class ServerInfo {

    private int tileSize;
    private TilesetMetadataInfo tileset;
    private SpritesMetadataInfo sprites;

    public ServerInfo(int tileSize, TilesetMetadataInfo tileset, SpritesMetadataInfo sprites) {
        this.tileSize = tileSize;
        this.tileset = tileset;
        this.sprites = sprites;
    }

    public int getTileSize() {
        return tileSize;
    }

    public TilesetMetadataInfo getTileset() {
        return tileset;
    }

    public SpritesMetadataInfo getSprites() {
        return sprites;
    }
}

File: client/src/app/shared/server/server-info.ts
import {SpritesInfo} from "./sprites-info";
import {TilesetInfo} from "./tileset-info";

/**
 * Class that contains the response from a server information request.
 */
export class ServerInfo {

    tileSize: number;
    tileset: TilesetInfo;
    sprites: SpritesInfo;
}